vizro-core/tests/unit/vizro/models/_components/form/test_range_slider.py

```diff
@@ -292,81 +308,3 @@ def test_range_slider_build_with_optional(self, expected_range_slider_with_optio
         expected = json.loads(json.dumps(expected_range_slider_with_optional, cls=plotly.utils.PlotlyJSONEncoder))
 
         assert result == expected
-
-
-class TestCallbackMethod:
```

Comment:
:(

Reply:
I know :/

Still, if we move javascript code into separate `statis/js/` files, it seems feasible to write unit tests for these functions as well.

Reply:
We may want to go this route: https://jest-community.github.io/jest-pytest/docs/overview.html

Reply:
I found several ways to test javascript functions. 
- Option 1 - `pytest`:
    - By using standard python lib called `subprocess` that triggers pure javascript test functions without any javascript test frameworks. 
    - Faster to implement and incorporate into existing project and CI process.
    - We don’t need to learn a new javascript framework
    - We can utilise all pytest options like test parametrisation.
    - But it’s a bad practice (hacky way to test it) especially for the open sourced project.
- Option 2 - `jest`:
    - By using jest javascript framework.
    - Not-to-easy to implement and incorporate into existing project and CI process (Node.js & npm have to be installed and configured).
    - We need to learn a new js framework.
    - But it’s a good practice especially for the open sourced project.
- Option 3 - pytest-jest:
    - The package has 40 stars in github and no documentation.

My preference is the second option because it's the only one that looks reliable and can meet the needs of the project (especially once the number of clientside callback functions grows over time). 
In that case I'd leave these changes for the separate PR, because the `node.js` & `npm` have to be installed, configured locally and for the CI process. However, I guess there is no big rush for it since these callbacks are already tested with UI tests (If I'm correct @l0uden)?

Reply:
`However, I guess there is no big rush for it since these callbacks are already tested with UI tests (If I'm correct @l0uden)?`

Sure, integration tests cover working of filter, parameters, interactions etc. on the UI level.

Reply:
Thanks a lot @petar-qb for the summary. I am still undecided...

Option 1 - the big advantage I see is that we do not leave our python first approach, and we could continue to cover everything with `hatch` 

Option 2 - I agree that this is the clean way to do it, but we would also loose a lot, by not relying on hatch alone to manage environments.

Option 3 - Agreed, let's not do that.

Can you maybe to a quick spike on option 2 that checks (no need to run complicated tests, just a dummy one):
- is it possible to create a command `hatch run test-js` that either runs those tests OR tells you exactly what you need to install how to get it run
- check to what degree this crosses with `hatch` in general

Reply:
@maxschulz-COL here is a first version of javascript testing setup. I said first version because probably the output from `hatch run test-js` will be changed 😄. 

I have one UX improvement and will configure it if you agree with. If the all the Node.js/npm setup steps are successfully configured, does it make sense to run js tests on `hatch run test-js`. And if tests can't be run, that `hatch run test-js` shows guidelines on how to setup it? 

I have one user experience improvement and I will configure it if you agree. If all Node.js/npm setup steps are successfully configured, does it make sense to run js tests on `hatch run test-js`. And if the tests can't run, does that `hatch run test-js` show guidance on how to set it up?

Reply:
Yes, that is how I imagine it, if that's possible

Reply:
Ok I followed the guide, and I was easily able to run the tests I think. Now I think we should just make it so that `hatch run test-js` runs the tests if your setup is working.

If the hatch script becomes unwiedy, we may want to create a script for this!

Reply:
I just started to add separate bash script for handling it inside the `vizro-core/tools`.